flycheck.el

```diff
@@ -7721,6 +7721,20 @@ Otherwise return the previously used cache directory."
         (or flycheck-haskell-ghc-cache-directory
             (make-temp-file "flycheck-haskell-ghc-cache" 'directory))))
 
+(defun flycheck--locate-dominating-file-matching (start-dir regexp)
+  "Search for a file in directory hierarchy starting at START-DIR.
+
+Look up the directory hierarchy from START-DIR for a directory
+containing file that matches file that matches REGEXP."
+  (locate-dominating-file
+   start-dir
+   (lambda (dir)
+     (directory-files dir
+                      nil ;; use full paths
```

Comment:
`nil` does *not* use full paths, according to the doc I have.  Besides, I'm not sure the comments are necessary when you can lookup the docstring.